djconnectwise/sync.py

```diff
@@ -1384,8 +1389,13 @@ def _assign_field_data(self, instance, json_data):
             )
 
         try:
-            member_id = json_data['member']['id']
+            member_id = json_data.get('member').get('id')
             models.Member.objects.get(pk=member_id)
+        except AttributeError:
+            raise InvalidObjectException(
+                'Schedule entry {} can not be found with no member info: '
```

Comment:
A schedule entry can exist without a member, we just don't currently track those.

Can you update the comment to say "Schedule entry {} does not have a related member: {} - skipping."

Also can you add a comment as to why it's ok to call `.get()` on the return value of `.get()` here. That's normally not a great idea but here it is ok since we are using the pythonic "try it first" method and we know we don't care about any shcedule entries without members.

Reply:
Maybe you seems to think about the case of block from lines 1399, `ObjectDoesNotExist`. I saw the error case of `json_data.get('member')` is None. 

> "Schedule entry {} does not have a related member: {} - skipping."

So, we don't know about the 2nd {}, member info. 
```
        except AttributeError:
            # AttributeError from 'member' or 'id' is the same, so subsequent
            # get() is okay here.
            raise InvalidObjectException(
                'Schedule entry {}: No member info exists. '
                '- skipping.'.format(instance.id)
            )
```

Reply:
Yes I didn't mean to include the second `{}`

If id returned `None` it wouldn't raise an AttributeError, it would raise a `DoesNotExist` exception.

Reply:
I mean the case that 'member' is None. In that case, can we meet AttributeError when calling `get('id')`?

> json_data.get('member') is None.

![image](https://user-images.githubusercontent.com/73486314/140585269-8aa583c6-844e-4aa5-b252-0b2f69fbe2f5.png)

Reply:
Actually I didn't see this from the gitlab view, but when I opened the file you can see that on line 1385 we already raise an exception if there is no member data to block schedule entries with no member, so this AttributeError block will never run. 

You can remove this AttributeError block and replace the assignment with:

```
member_id = json_data['member'].get('id') 
```

Reply:
Yeap, member is being checked in the above block. We can keep simpler codes. Thanks.

2 fixes you mentioned are applied, and ready for review.